src/communication/xcWebSocketPublisher.js

```diff
@@ -34,6 +33,35 @@ define(function () {
     }
 
 
+    Publisher.prototype.getEventToSendContext = function (stateMachineRef, jsonMessage) {
+        var componentCode = stateMachineRef.ComponentCode.Fields[0];
+        var stateMachineCode = stateMachineRef.StateMachineCode.Fields[0];
+        var publish = this.configuration.getPublisherDetails(componentCode, stateMachineCode);
+        var event = {
+            "Header": {
+                "AgentId": stateMachineRef.AgentId,
+                "StateMachineId": stateMachineRef.StateMachineId,
+                "StateMachineCode": stateMachineRef.StateMachineCode,
+                "ComponentCode": stateMachineRef.ComponentCode,
+                "EventCode": parseInt(publish.eventCode),
+                "IncomingType": 0,
+                "MessageType": { "Case": "Some", "Fields": [publish.messageType] }
+            },
+            "JsonMessage": JSON.stringify(jsonMessage)
+        };
+        return {
+            event: event,
+            routingKey: publish.routingKey
+        };
+    }
+
+
+    Publisher.prototype.sendContext = function (stateMachineRef, jsonMessage) {
```

Comment:
find a better name. (use statemachineref instead of context)
factorize your code (avoid code duplication)